openapi/tests/openapi_integration/test_collection_update.py

```diff
@@ -19,12 +19,16 @@ def test_collection_update():
         method="PATCH",
         path_params={'collection_name': collection_name},
         body={
+            "vectors": {
+                "hnsw": {
+                    "m": 16,
+                },
+            },
             "optimizers_config": {
                 "default_segment_number": 6,
                 "indexing_threshold": 10000,
             },
             "hnsw": {
-                "m": 42,
                 "ef_construct": 123,
```

Comment:
I propose to move `ef_construct` into `vectors` too

Reply:
The idea was that this way we have both a collection and vector-specific HNSW configuration, which creates a situation where both must be merged.

The effective configuration isn't actually tested here however.

I'll simplify this test as you've suggested. I'll also add a new test to actually test the effective configuration.